tls/extensions/s2n_client_supported_groups.h

```diff
@@ -15,10 +15,13 @@
 
 #pragma once
 
+#include "tls/extensions/s2n_extension_type.h"
 #include "tls/s2n_connection.h"
 #include "stuffer/s2n_stuffer.h"
 
-extern int s2n_extensions_client_supported_groups_send(struct s2n_connection *conn, struct s2n_stuffer *out);
-extern int s2n_recv_client_supported_groups(struct s2n_connection *conn, struct s2n_stuffer *extension);
-int s2n_parse_client_supported_groups_list(struct s2n_connection *conn, struct s2n_blob *iana_ids, const struct s2n_ecc_named_curve **supported_groups);
-int s2n_choose_supported_group(struct s2n_connection *conn, const struct s2n_ecc_named_curve **group_options, struct s2n_ecc_evp_params *chosen_group);
```

Comment:
Why exactly are you removing this function definition from the header file?

Reply:
*this function meaning s2n_choose_supported_group()

Reply:
It's only used in the unit tests, not in any other part of the code base, so it doesn't need to be advertised in the header. It's an implementation detail.